R/idmc_transform_daily.R

```diff
@@ -45,20 +66,20 @@ idmc_transform_daily <- function(df) {
       !is.na(.data[["displacement_date"]]) # drop where no start/end available
     ) %>%
     dplyr::mutate(
-      start_date = dplyr::if_else( # reverse and fill start_date if necess
+      start_date = dplyr::if_else( # reverse and fill start_date if necessary
         is.na(.data[[e_col_]]) | .data[[s_col_]] <= .data[[e_col_]],
         .data[[s_col_]],
         .data[[e_col_]]
       ),
-      end_date = dplyr::if_else( # reverse and fill end date if necess
+      end_date = dplyr::if_else( # reverse and fill end date if necessary
         !is.na(.data[[e_col_]]) & .data[[s_col_]] <= .data[[e_col_]],
         .data[[e_col_]],
         .data[[s_col_]]
       )
     )
 
   # create daily displacement from events
-  df_correct %>%
+  df_daily <- df_correct %>%
     dplyr::rowwise() %>%
     dplyr::mutate(
       date = list(
```

Comment:
is there a reason why you sometimes use  either`$`  or `[[ ]]` in the context of this function?

Reply:
Yeah, in this case, it was because of the extremely long variable names above that I was using the `[[s_col_]]` format rather than `.data$displacement_start_date`. I generally avoid the string format as much as possible unless I need to dynamically pull in a string after pivoting, otherwise I need to parse a string to column, or a user needs to more easily pass a string and somehow it can't be `{{ col }}`. This was basically for code readability solely ha!